packages/metro/src/DeltaBundler/graphOperations.js

```diff
@@ -450,6 +459,10 @@ function removeDependency<T>(
     decrementImportBundleReference(dependency, graph);
   }
 
+  if (dependency.data.data.contextParams) {
+    graph.privateState.resolvedContext.delete(dependency.absolutePath);
```

Comment:
This isn't the right place for this - it should happen in `releaseModule`. That's a bug, because if there's a reference to the same context elsewhere in the graph, we'd end up deleting it while it's still in use.